jastrow.py

```diff
@@ -266,47 +360,80 @@ def _get_deltab(self,e,epos):
         """
         nconf=epos.shape[0]
         ne=self._configscurrent.shape[1]
+        nup = self._mol.nelec[0]
         mask=[True]*ne
         mask[e]=False
 
-        dnew=eedist_i(self._configscurrent,epos)[:,mask,:].reshape((-1,3))
-        dold=eedist_i(self._configscurrent,self._configscurrent[:,e,:])[:,mask,:].reshape((-1,3))
-        delta=np.zeros((nconf,len(self.b_basis)))
+        dnew=eedist_i(self._configscurrent,epos)[:,mask,:]
+        dold=eedist_i(self._configscurrent,self._configscurrent[:,e,:])[:,mask,:]
+
+        if(e < nup): # Spin up electron selected
```

Comment:
Same things about the if statements here; you don't need to have them and the code ends up twice as long as it needs to be.

Reply:
Similarly precomputed the if statement condition and used it as an index